src/Sarif.Converters/SemmleQlConverter.cs

```diff
@@ -221,6 +221,11 @@ private IList<Location> NormalizeRawMessage(string rawMessage, out string normal
                         };
                     }
 
+                    if (physicalLocation.Region.ByteOffset == 0)
```

Comment:
>ByteOffset [](start = 48, length = 10)

this doesn't look right to me. isn't it possible that Semmle sets a byteOffset to 0? I see parsing occurring above. Since -1 is already the default, why do you have to set it here? I'd expect byteOffset to be -1 on instantiating the region. if the code above replaces it, then the default is overridden for a useful purpose. Ask Larry his opinion here.

Reply:
i updated this to use ByteLength instead. The reason i need to set it here is that incoming object has a default value of zero and if we dont override it, we end up getting zero valued ByteOffsets. I think having the check against ByteLength is a good check.

---
In reply to: [277867847](https://github.com/Microsoft/sarif-sdk/pull/1451#discussion_r277867847) [](ancestors = 277867847)